lightning/src/ln/channelmanager.rs

```diff
@@ -2787,16 +2796,16 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 									 HTLCFailReason::Reason { failure_code: 0x4000|15, data: htlc_msat_height_data });
 				} else {
 					match self.claim_funds_from_hop(channel_state.as_mut().unwrap(), htlc.prev_hop, payment_preimage) {
-						Err(Some(e)) => {
-							if let msgs::ErrorAction::IgnoreError = e.1.err.action {
+						ClaimFundsFromHop::MonitorUpdateFail(pk, err, _) => {
+							if let msgs::ErrorAction::IgnoreError = err.err.action {
 								// We got a temporary failure updating monitor, but will claim the
 								// HTLC when the monitor updating is restored (or on chain).
-								log_error!(self.logger, "Temporary failure claiming HTLC, treating as success: {}", e.1.err.err);
+								log_error!(self.logger, "Temporary failure claiming HTLC, treating as success: {}", err.err.err);
 								claimed_any_htlcs = true;
-							} else { errs.push(e); }
+							} else { errs.push((pk, err)); }
 						},
-						Err(None) => unreachable!("We already checked for channel existence, we can't fail here!"),
-						Ok(()) => claimed_any_htlcs = true,
+						ClaimFundsFromHop::PrevHopForceClosed => unreachable!("We already checked for channel existence, we can't fail here!"),
+						_ => claimed_any_htlcs = true,
 					}
```

Comment:
Document that this function may return `true` for duplicate claims?

Reply:
Actually, I changed it not to instead.